protocol/synthetix/contracts/storage/CollateralConfiguration.sol

```diff
@@ -212,20 +217,39 @@ library CollateralConfiguration {
 
     /**
      * @dev Converts token amounts with non-system decimal precisions, to 18 decimals of precision.
-     * E.g: USDC uses 6 decimals of precision, so this would upscale it by 12 decimals.
+     * E.g: $TOKEN_A uses 6 decimals of precision, so this would upscale it by 12 decimals.
+     * E.g: $TOKEN_B uses 20 decimals of precision, so this would downscale it by 2 decimals.
      * @param self The CollateralConfiguration object corresponding to the collateral type being converted.
      * @param tokenAmount The token amount, denominated in its native decimal precision.
-     * @return The converted amount, denominated in the system's 18 decimal precision.
+     * @return amountD18 The converted amount, denominated in the system's 18 decimal precision.
      */
     function convertTokenToSystemAmount(
         Data storage self,
         uint256 tokenAmount
-    ) internal view returns (uint256) {
+    ) internal view returns (uint256 amountD18) {
         // this extra condition is to prevent potentially malicious untrusted code from being executed on the next statement
         if (self.tokenAddress == address(0)) {
             revert CollateralNotFound();
         }
 
-        return (tokenAmount * DecimalMath.UNIT) / (10 ** IERC20(self.tokenAddress).decimals());
+        /// @dev this try-catch block assumes there is no malicious code in the token's fallback function
+        try IERC20(self.tokenAddress).decimals() returns (uint8 decimals) {
+            if (decimals == 18) {
+                amountD18 = tokenAmount;
+            } else if (decimals < 18) {
+                amountD18 = (tokenAmount * DecimalMath.UNIT) / (10 ** decimals);
+            } else {
+                // ensure no precision is lost when converting to 18 decimals
+                if (tokenAmount % (10 ** (decimals - 18)) != 0) {
+                    revert PrecisionLost();
```

Comment:
this is a very minor thing but its usually a good idea to include as much data as possible in revert reasons. so for `PrecisionLost`, including the `tokenAmount` and the number of decimals reported by the token would be potentially useful in case a confused developer is debugging this code and not understanding why it is erroring exactly